app/src/main/java/org/dhis2/Bindings/TrackedEntityInstanceExtensions.kt

```diff
@@ -25,3 +26,28 @@ fun MutableList<TrackedEntityInstance>.filterDeletedEnrollment(
 
     return this
 }
+
+fun MutableList<TrackedEntityInstance>.filterEnrollmentStatus(
+    d2: D2,
+    program: String?,
+    enrollmentStatuses: List<EnrollmentStatus>
+): MutableList<TrackedEntityInstance> {
+    val iterator = this.iterator()
+    if (program != null && enrollmentStatuses.isNotEmpty()) {
+        while (iterator.hasNext()) {
+            val tei = iterator.next()
+            val hasEnrollmentInProgramWithStatus =
+                !d2.enrollmentModule().enrollments()
+                    .byTrackedEntityInstance().eq(tei.uid())
+                    .byProgram().eq(program)
+                    .byDeleted().isFalse
+                    .byStatus().`in`(enrollmentStatuses)
+                    .blockingIsEmpty()
+            if (!hasEnrollmentInProgramWithStatus) {
+                iterator.remove()
+            }
+        }
+    }
+
+    return this
+}
```

Comment:
I miss unit tests :dancers: . Also, I'd say careful with having statics in memory :+1: . This more than a "utils" method is a use case.

Reply:
Unit test done!